Description: Fix firefox profile
 Adjust firefox profile to properly work sandboxed for daily use.
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

--- firejail-0.9.54.orig/etc/firefox-common-addons.inc
+++ firejail-0.9.54/etc/firefox-common-addons.inc
@@ -61,4 +61,4 @@ noblacklist /usr/lib/python3*
 
 # Flash plugin
 # private-etc must first be enabled in firefox-common.profile and in profiles including it.
-#private-etc adobe
+private-etc adobe
--- firejail-0.9.54.orig/etc/firefox-common.profile
+++ firejail-0.9.54/etc/firefox-common.profile
@@ -16,7 +16,14 @@ include /etc/firejail/disable-interprete
 include /etc/firejail/disable-programs.inc
 
 mkdir ${HOME}/.pki
+
 whitelist ${DOWNLOADS}
+whitelist ${HOME}/Documents
+whitelist ${HOME}/Desktop
+whitelist ${HOME}/Pictures
+whitelist ${HOME}/Music
+whitelist ${HOME}/Video
+
 whitelist ${HOME}/.pki
 include /etc/firejail/whitelist-common.inc
 include /etc/firejail/whitelist-var-common.inc
@@ -39,9 +46,10 @@ shell none
 #tracelog
 
 disable-mnt
-private-dev
+#disabled because of u2f devices support
+#private-dev
 # private-etc below works fine on most distributions. There are some problems on CentOS.
-#private-etc ca-certificates,ssl,machine-id,dconf,selinux,passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,xdg,gtk-2.0,gtk-3.0,X11,pango,fonts,mime.types,mailcap,asound.conf,pulse,pki,crypto-policies,ld.so.cache
+private-etc ca-certificates,ssl,machine-id,dconf,selinux,passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,xdg,gtk-2.0,gtk-3.0,X11,pango,fonts,mime.types,mailcap,asound.conf,pulse,pki,crypto-policies,ld.so.cache,alternatives
 private-tmp
 
 noexec ${HOME}
